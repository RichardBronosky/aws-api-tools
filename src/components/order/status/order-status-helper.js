import { getOrderList } from '@services/order-status'
import { announce } from '@helpers/aria-announce'
import { validatePhone } from '@helpers/string-helper'

export const getRelatedOrders = (e, orderId, phoneNumber, loading, setState, setSearchState, ref) => {
  e.preventDefault()
  if (!loading) {
    setSearchState({ loading: true })
    const validPhone = validatePhone(phoneNumber)
    if (validPhone) {
      getOrderList({ orderId: orderId, phoneNumber: phoneNumber.replace(/[^0-9]/g, '') })
        .then(orderList => {
          setState({ orderList: orderList })
          ref && window.scrollTo(0, ref.current.clientHeight)
          setSearchState({ err: null, loading: false })
        })
        .catch(err => {
          const error =
            err && err.message && err.message.data && err.message.data.error && err.response.data.error.message
              ? err.response.data.error.message
              : 'We are unable to retrieve your order, please try again later.'
          announce(error)
          setState({ orderList: {} })
          setSearchState({ err: error, loading: false })
        })
    } else {
      const error = 'Invalid phone number.'
      announce(error)
      setState({ orderList: {} })
      setSearchState({ err: error, loading: false })
    }
  }
}

export const viewOrderList = setState => {
  setState({ orderDetails: null })
}

export const viewOrderDetails = (order, setState) => {
  setState({ orderDetails: order })
  window.scrollTo(0, 0)
}

export const getFinDepositTotal = deposits => {
  let total = 0
  if (deposits.length > 0) {
    for (let i = 0, n = deposits.length; i < n; i++) {
      total += deposits[i].AuthorizedAmount
    }
  }
  return total
}

export const getOtherPayments = order => {
  let otherPayments = []
  if (order) {
    if (order.Deposits && order.Deposits.Deposit.length > 0) {
      otherPayments = [...otherPayments, ...order.Deposits.Deposit]
    }
    if (order.GCDeposits && order.GCDeposits.Deposit.length > 0) {
      otherPayments = [...otherPayments, ...order.GCDeposits.Deposit]
    }
  }
  return otherPayments
}

export const getPaymentType = dep => {
  switch (dep.CashCode) {
    case 'MC':
      return 'Master Card'
    case 'VA':
      return 'Visa'
    case 'DV':
      return 'Discover'
    case 'AMX':
      return 'American Express'
    case 'GC':
      return 'Gift Card'
    case 'PAL':
      return 'PayPal'
    case 'AFF':
      return 'Affirm'
  }
}
