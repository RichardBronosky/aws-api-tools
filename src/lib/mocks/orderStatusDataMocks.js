export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
