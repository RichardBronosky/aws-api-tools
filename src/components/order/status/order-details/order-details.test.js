import React from 'react'
import renderer, { act } from 'react-test-renderer'
import OrderDetails from '../order-details'
import { orderStatus } from '@mocks/orderStatusDataMocks'
import * as orderStatusHelper from '../order-status-helper'

describe('OrderDetails', () => {
  const testProps = {
    order: orderStatus,
    setOrderStatusState: jest.fn(),
  }

  const setState = jest.fn()
  let useStateSpy

  beforeEach(() => {
    useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation(init => [init, setState])
  })

  afterAll(() => {
    useStateSpy.mockRestore()
  })

  it('renders correctly when order exists', () => {
    const tree = renderer.create(<OrderDetails { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when order exists and payment details expanded', () => {
    const spy = jest.spyOn(React, 'useState').mockImplementation(() => [true, jest.fn()])
    const tree = renderer.create(<OrderDetails { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
    spy.mockRestore()
  })

  it("renders correctly when order doesn't have other payment methods", () => {
    const tree = renderer.create(
      <OrderDetails { ...{ ...testProps, order: { ...testProps.order, GCDeposits: null, Deposits: null } } } />
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('calls viewOrderList on back to orders click', () => {
    const tree = renderer.create(<OrderDetails { ...testProps } />)
    orderStatusHelper.viewOrderList = jest.fn()
    tree.root.findByProps({ className: 'primary' }).props.onClick()
    expect(orderStatusHelper.viewOrderList).toHaveBeenCalled()
  })

  it('updates state when clicking on other payment details expander button', () => {
    const tree = renderer.create(<OrderDetails { ...testProps } />)
    tree.root.findByProps({ className: 'other-payment-details-expander' }).props.onClick()
    expect(setState).toHaveBeenCalled()
  })

  it('updates state when clicking on finance details expander button', () => {
    const tree = renderer.create(<OrderDetails { ...testProps } />)
    tree.root.findByProps({ className: 'finance-details-expander' }).props.onClick()
    expect(setState).toHaveBeenCalled()
  })
})
