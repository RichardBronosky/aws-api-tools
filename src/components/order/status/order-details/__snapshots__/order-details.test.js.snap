// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails renders correctly when order doesn't have other payment methods 1`] = `
<div
  className="grid-x cell small-12 order-details"
>
  <div
    className="grid-x cell small-12 order-status-detailed-header"
  >
    <div
      className="cell small-12 medium-6 large-6 order-number"
    >
      <h2>
        Order # 14000136I
      </h2>
      <p
        className="ordered-on"
      >
        Ordered on Sunday, August 04, 2019
      </p>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <button
        className="primary"
        onClick={[Function]}
      >
        &lt; Back to Orders
      </button>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-payment-history"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Payment History
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="payment-section"
      >
        <div
          className="cell small-12"
        >
          <strong>
            RTG Finance
          </strong>
        </div>
        <div
          className="cell small-12"
        >
          $600
           (
          <button
            aria-controls="finance-details-accordion"
            aria-expanded={false}
            className="finance-details-expander"
            onClick={[Function]}
            tabIndex="0"
          >
            see details
          </button>
          )
        </div>
        <div
          id="finance-details-accordion"
        />
      </div>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <span>
        <strong>
          Note: 
        </strong>
        Delivery dates on orders with an open balance are subject to change. Call Customer Care if you need help at 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
        .
      </span>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-delivery-setup"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Delivery & Setup
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="cell small-12 bold"
      >
        <strong>
          SHIPPED TO
        </strong>
      </div>
      <div
        className="cell small-12"
      >
        <p>
          SARAH
           
          SMITH
        </p>
        <div
          className="order-status-delivery-address"
        >
          <p>
            1112 MARSH TRAIL CIR
          </p>
          <p>
            
          </p>
          <p>
            ATLANTA
            , 
            GA
            , 
            30328
          </p>
        </div>
      </div>
    </div>
    <div
      className="cell small-12 medium-6"
    >
      <strong>
        Scheduled Delivery Date
      </strong>
      <p>
        Contact Customer Service to Schedule a Delivery Date - 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
      </p>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-product-info"
  >
    <div
      className="grid-x cell small-12 order-product-info-list"
    >
      <div
        className="grid-x cell small-12 order-status-product-info-headings-main"
      >
        <div
          className="order-status-heading cell small-12 medium-4 large-7"
        >
          Product Information
        </div>
        <div
          className="order-status-heading cell small-12 medium-2 large-1"
        >
          Qty.
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Price
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Subtotal
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-product-info-items"
      >
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Louie Gray Twin Loft Bed with Desk and Bookcase
              <p>
                SKU: 3611163P
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $444
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $444
            </div>
          </div>
        </div>
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Mattie Gray Chevron Bean Bag Chair
              <p>
                SKU: 38096613
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $29
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $29
            </div>
          </div>
        </div>
      </div>
      <div
        className="order-status-product-info-totals grid-x cell small-12"
      >
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Sub Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $518.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Delivery & Shipping
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $119.99
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Tax
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $45.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $638.95
          </div>
        </div>
        <div
          className="grid-x cell small-12 balance"
        >
          <div
            className="cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            <strong>
              Balance:
            </strong>
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            <strong>
              $0
            </strong>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderDetails renders correctly when order exists 1`] = `
<div
  className="grid-x cell small-12 order-details"
>
  <div
    className="grid-x cell small-12 order-status-detailed-header"
  >
    <div
      className="cell small-12 medium-6 large-6 order-number"
    >
      <h2>
        Order # 14000136I
      </h2>
      <p
        className="ordered-on"
      >
        Ordered on Sunday, August 04, 2019
      </p>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <button
        className="primary"
        onClick={[Function]}
      >
        &lt; Back to Orders
      </button>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-payment-history"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Payment History
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="payment-section"
      >
        <div
          className="cell small-12"
        >
          <strong>
            Other Payments
          </strong>
        </div>
        <div
          className="cell small-12"
        >
          $38.95
           
          (
          <button
            aria-controls="other-payment-details-accordion"
            aria-expanded={false}
            className="other-payment-details-expander"
            onClick={[Function]}
            tabIndex="0"
          >
            see details
          </button>
          )
        </div>
        <div
          id="other-payment-details-accordion"
        />
      </div>
      <div
        className="payment-section"
      >
        <div
          className="cell small-12"
        >
          <strong>
            RTG Finance
          </strong>
        </div>
        <div
          className="cell small-12"
        >
          $600
           (
          <button
            aria-controls="finance-details-accordion"
            aria-expanded={false}
            className="finance-details-expander"
            onClick={[Function]}
            tabIndex="0"
          >
            see details
          </button>
          )
        </div>
        <div
          id="finance-details-accordion"
        />
      </div>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <span>
        <strong>
          Note: 
        </strong>
        Delivery dates on orders with an open balance are subject to change. Call Customer Care if you need help at 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
        .
      </span>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-delivery-setup"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Delivery & Setup
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="cell small-12 bold"
      >
        <strong>
          SHIPPED TO
        </strong>
      </div>
      <div
        className="cell small-12"
      >
        <p>
          SARAH
           
          SMITH
        </p>
        <div
          className="order-status-delivery-address"
        >
          <p>
            1112 MARSH TRAIL CIR
          </p>
          <p>
            
          </p>
          <p>
            ATLANTA
            , 
            GA
            , 
            30328
          </p>
        </div>
      </div>
    </div>
    <div
      className="cell small-12 medium-6"
    >
      <strong>
        Scheduled Delivery Date
      </strong>
      <p>
        Contact Customer Service to Schedule a Delivery Date - 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
      </p>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-product-info"
  >
    <div
      className="grid-x cell small-12 order-product-info-list"
    >
      <div
        className="grid-x cell small-12 order-status-product-info-headings-main"
      >
        <div
          className="order-status-heading cell small-12 medium-4 large-7"
        >
          Product Information
        </div>
        <div
          className="order-status-heading cell small-12 medium-2 large-1"
        >
          Qty.
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Price
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Subtotal
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-product-info-items"
      >
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Louie Gray Twin Loft Bed with Desk and Bookcase
              <p>
                SKU: 3611163P
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $444
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $444
            </div>
          </div>
        </div>
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Mattie Gray Chevron Bean Bag Chair
              <p>
                SKU: 38096613
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $29
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $29
            </div>
          </div>
        </div>
      </div>
      <div
        className="order-status-product-info-totals grid-x cell small-12"
      >
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Sub Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $518.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Delivery & Shipping
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $119.99
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Tax
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $45.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $638.95
          </div>
        </div>
        <div
          className="grid-x cell small-12 balance"
        >
          <div
            className="cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            <strong>
              Balance:
            </strong>
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            <strong>
              $0
            </strong>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderDetails renders correctly when order exists and payment details expanded 1`] = `
<div
  className="grid-x cell small-12 order-details"
>
  <div
    className="grid-x cell small-12 order-status-detailed-header"
  >
    <div
      className="cell small-12 medium-6 large-6 order-number"
    >
      <h2>
        Order # 14000136I
      </h2>
      <p
        className="ordered-on"
      >
        Ordered on Sunday, August 04, 2019
      </p>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <button
        className="primary"
        onClick={[Function]}
      >
        &lt; Back to Orders
      </button>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-payment-history"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Payment History
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="payment-section"
      >
        <div
          className="cell small-12"
        >
          <strong>
            Other Payments
          </strong>
        </div>
        <div
          className="cell small-12"
        >
          $38.95
           
          (
          <button
            aria-controls="other-payment-details-accordion"
            aria-expanded={true}
            className="other-payment-details-expander"
            onClick={[Function]}
            tabIndex="0"
          >
            hide details
          </button>
          )
        </div>
        <div
          id="other-payment-details-accordion"
        >
          <div
            className="payment"
          >
            <div>
              <div
                className="cell small-12"
              >
                 
                8/4/2019
                 
              </div>
              <div
                className="cell small-12"
              >
                 
                Visa
                 
              </div>
              <div
                className="cell small-12"
              >
                $37
              </div>
            </div>
          </div>
          <div
            className="payment"
          >
            <div>
              <div
                className="cell small-12"
              >
                 
                8/4/2019
                 
              </div>
              <div
                className="cell small-12"
              >
                 
                Gift Card
                 
              </div>
              <div
                className="cell small-12"
              >
                $1.95
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="payment-section"
      >
        <div
          className="cell small-12"
        >
          <strong>
            RTG Finance
          </strong>
        </div>
        <div
          className="cell small-12"
        >
          $600
           (
          <button
            aria-controls="finance-details-accordion"
            aria-expanded={true}
            className="finance-details-expander"
            onClick={[Function]}
            tabIndex="0"
          >
            hide details
          </button>
          )
        </div>
        <div
          id="finance-details-accordion"
        >
          <div
            className="payment"
          >
            <div
              className="cell small-12"
            >
              8/4/2019
            </div>
            <div
              className="cell small-12"
            >
              $600
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cell small-12 medium-6 large-6"
    >
      <span>
        <strong>
          Note: 
        </strong>
        Delivery dates on orders with an open balance are subject to change. Call Customer Care if you need help at 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
        .
      </span>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-delivery-setup"
  >
    <div
      className="cell small-12"
    >
      <h3>
        Delivery & Setup
      </h3>
    </div>
    <div
      className="grid-x cell small-12 medium-6 large-6"
    >
      <div
        className="cell small-12 bold"
      >
        <strong>
          SHIPPED TO
        </strong>
      </div>
      <div
        className="cell small-12"
      >
        <p>
          SARAH
           
          SMITH
        </p>
        <div
          className="order-status-delivery-address"
        >
          <p>
            1112 MARSH TRAIL CIR
          </p>
          <p>
            
          </p>
          <p>
            ATLANTA
            , 
            GA
            , 
            30328
          </p>
        </div>
      </div>
    </div>
    <div
      className="cell small-12 medium-6"
    >
      <strong>
        Scheduled Delivery Date
      </strong>
      <p>
        Contact Customer Service to Schedule a Delivery Date - 
        <a
          href="tel:(800)-766-6786"
        >
          (800)-766-6786
        </a>
      </p>
    </div>
  </div>
  <div
    className="grid-x cell small-12 order-status-product-info"
  >
    <div
      className="grid-x cell small-12 order-product-info-list"
    >
      <div
        className="grid-x cell small-12 order-status-product-info-headings-main"
      >
        <div
          className="order-status-heading cell small-12 medium-4 large-7"
        >
          Product Information
        </div>
        <div
          className="order-status-heading cell small-12 medium-2 large-1"
        >
          Qty.
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Price
        </div>
        <div
          className="order-status-heading cell small-12 medium-3 large-2"
        >
          Subtotal
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-product-info-items"
      >
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Louie Gray Twin Loft Bed with Desk and Bookcase
              <p>
                SKU: 3611163P
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $444
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $444
            </div>
          </div>
        </div>
        <div
          className="grid-x cell small-12 order-product-info-list-item"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7 undefined"
            >
              Mattie Gray Chevron Bean Bag Chair
              <p>
                SKU: 38096613
              </p>
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              1
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              $29
            </div>
            <div
              className="order-status-heading subtotal cell small-12 medium-3 large-2"
            >
              $29
            </div>
          </div>
        </div>
      </div>
      <div
        className="order-status-product-info-totals grid-x cell small-12"
      >
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Sub Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $518.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Delivery & Shipping
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $119.99
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Tax
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $45.96
          </div>
        </div>
        <div
          className="grid-x cell small-12"
        >
          <div
            className="order-status-heading cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            Total
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            $638.95
          </div>
        </div>
        <div
          className="grid-x cell small-12 balance"
        >
          <div
            className="cell large-8"
          />
          <div
            className="order-status-heading cell small-9 large-2"
          >
            <strong>
              Balance:
            </strong>
          </div>
          <div
            className="order-status-heading cell small-3 large-2"
          >
            <strong>
              $0
            </strong>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
